### PR TITLE
[eclipse/xtext#1384] Change Jenkinsfile to Declarative Pipeline

### PR DESCRIPTION
**Pull request by kthoms**
- externalized build steps into shell scripts for convenient local usage
- added Slack notifications
- configure Maven default goal

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
+
+	post {
+		success {
+			archiveArtifacts artifacts: 'build/**'
+		}
+		failure {
+			slackSend baseUrl: 'https://itemis.slack.com/services/hooks/jenkins-ci/', botUser: true, channel: 'xtext-builds', color: '#FF0000', message: "FAILED: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}>", token: '1vbkhv8Hwlp3ausuFGj1BdJb'
+		}
+		regression {
+			slackSend baseUrl: 'https://itemis.slack.com/services/hooks/jenkins-ci/', botUser: true, channel: 'xtext-builds', color: '#FF0000', message: "FAILED: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}>", token: '1vbkhv8Hwlp3ausuFGj1BdJb'
```
Comment (cdietrich) [inline Jenkinsfile]: do you already know if this can be done via env as well?

**Comment by cdietrich** [comment]
fails here: http://services.typefox.io/open-source/jenkins/job/xtext-core/job/kth_issue1384/

**Comment by kthoms** [comment]
Works now both on JIPP & TF Jenkins

File: Jenkinsfile (hunk 1 above)
Comment (kthoms) [reply Jenkinsfile]: current version only uses `changed` and scripts within that. This is portable for both Jenkins versions.

**Comment by cdietrich** [comment]
@kthoms can this PR be closed?

**Pull request merged**